From e328f510e0fa49b0661bc2ed368add4553362175 Mon Sep 17 00:00:00 2001
From: Kai Liu <kai.liu@suse.com>
Date: Mon, 27 Jun 2022 13:33:46 +0800
Subject: [PATCH] Revert "bpf: Fix incorrect memory charge cost calculation in
 stack_map_alloc()"
Patch-mainline: Never, revert as we have backported mainline version
References: bsn#83
Modified-by-SEL: No

This reverts commit 28bbdca6a7a471921d890e5c0d70b6f7c99637a7.
---
 kernel/bpf/stackmap.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/bpf/stackmap.c b/kernel/bpf/stackmap.c
index 0c5bf98d5576..c19e669afba0 100644
--- a/kernel/bpf/stackmap.c
+++ b/kernel/bpf/stackmap.c
@@ -121,8 +121,7 @@ static struct bpf_map *stack_map_alloc(union bpf_attr *attr)
 		return ERR_PTR(-E2BIG);
 
 	cost = n_buckets * sizeof(struct stack_map_bucket *) + sizeof(*smap);
-	err = bpf_map_charge_init(&mem, cost + attr->max_entries *
-			   (sizeof(struct stack_map_bucket) + (u64)value_size));
+	err = bpf_map_charge_init(&mem, cost);
 	if (err)
 		return ERR_PTR(err);
 
-- 
2.35.1

